src/web/core/src/main/java/org/geoserver/web/data/resource/BasicResourceConfig.java

```diff
@@ -105,13 +93,14 @@ public BasicResourceConfig(String id, IModel<ResourceInfo> model) {
         refForm.add(latLonPanel);
         refForm.add(computeLatLonBoundsLink(refForm, nativeBBox, latLonPanel));
 
-        // check other native srs for Cascaded WFS or WMS resource
-        List<String> otherSRS = getOtherSRS(model.getObject());
-
-        // native srs , declared srs, and srs handling dropdown
+        // ALERT - CORE CHANGE
+        // for resources coming from WMS and WFS Datastore
+        // then check if it has more than one SRS, then add FIND SRS button in Native SRS
+        // which will allow user to set other SRS as Native SRS
+        List<String> otherSRS = getOtherSRSFromWFSOrWMS(model.getObject());
         CRSPanel nativeCRS;
-
         if (otherSRS.isEmpty()) {
+            // legacy behavior
```

Comment:
Why calling this legacy? the way I see it there is the behavior for normal layers, and the behavior for cascaded WMS and WFS stores no?

Reply:
will change